### PR TITLE
docs: add commit and branching naming conventions to contributing.md

### PR DESCRIPTION
**Pull request by rudrakshkarpe**
## Description
Add commit and branching naming conventions to contributing.md

## Changes Made
Improved CONTRIBUTING.md with naming conventions

## Checklist
- [X] I have read and followed the contributing guidelines.
- [X] I have tested my changes thoroughly and they work as expected.
- [ ] I have added necessary tests for the changes made.
- [X] I have updated the documentation to reflect the changes made.
- [X] My code follows the project's coding style and standards.
- [X] I have added appropriate commit messages and comments for my changes.

## Related Issues
#24 Include commit message and branch naming conventions in the contributing.md

## Additional Notes
I am looking forward to contributing to the GCP Scanner!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,3 +26,21 @@ information on using pull requests.
...
+**Example:** Add support for GCP Compute Engine in scanner
+
+**Example:** feat:add support for GCP Compute Engine in scanner
+
+## Branching Naming Conventions
+- Use lowercase letters.
+- Separate words with hyphens.
+- Use descriptive names that reflect the purpose of the branch.
+
+**Example:** feature/gcp-compute-engine-support
```
Comment (sudiptob2) [inline CONTRIBUTING.md:46]:
```suggestion
**Example:** feat/<issue-id>/gcp-compute-engine-support
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -26,3 +26,21 @@ information on using pull requests.
...
+
+**Example:** Add support for GCP Compute Engine in scanner
+
+**Example:** feat:add support for GCP Compute Engine in scanner
+
+## Branching Naming Conventions
+- Use lowercase letters.
+- Separate words with hyphens.
+- Use descriptive names that reflect the purpose of the branch.
+
```
Comment (sudiptob2) [inline CONTRIBUTING.md:45]:
```suggestion
- Add `issue tracker id`

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -26,3 +26,21 @@ information on using pull requests.
...
+## Commit Message Conventions
+- Start with a short summary (50 characters or less) of the changes made.
+- Use the present tense and imperative mood.
+- Separate the summary from the body of the message with a blank line.
+- Use the body to explain what and why changes were made, as well as any necessary details.
+- Addionally, you can consider using semantic commit messages like "feat:" or "docs:" which will provide additional context to the commit message.
+
+**Example:** Add support for GCP Compute Engine in scanner
+
+**Example:** feat:add support for GCP Compute Engine in scanner
```
Comment (sudiptob2) [inline CONTRIBUTING.md:39]:
```suggestion
**Example:** feat: add support for GCP Compute Engine in scanner
```

**Review by sudiptob2 — CHANGES_REQUESTED** [review]
Seems like a good start. I think we need to provide a list of issue type. For starters,

- docs
- feat
- fix
- test
- ci
- chore
etcetera

**Comment by rudrakshkarpe** [comment]
> Seems like a good start. I think we also need to provide a list of the issue type. For starters,
> 
> - docs
> - feat
> - fix
> - test
> - ci
> - chore
> etcetera

Alright! I'll modify the PR with additional semantic commit messages. Thanks for reviewing it.

**Comment by mshudrak** [comment]
Same here. Please accept Google CLA first.

**Comment by rudrakshkarpe** [comment]
I have made the required changes and have signed the CLA. Could you please take a look at the changes and let me know if they meet the project's standards? I'm hoping to get my changes merged soon. Thank you!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -32,12 +32,24 @@ Guidelines](https://opensource.google/conduct/).
 - Use the present tense and imperative mood.
 - Separate the summary from the body of the message with a blank line.
 - Use the body to explain what and why changes were made, as well as any necessary details.
-- Addionally, you can consider using semantic commit messages like "feat:" or "docs:" which will provide additional context to the commit message.
+- Addionally, you can consider using semantic commit messages like "feat:" or "docs:" , etc. which will provide additional context to the commit message. Learn more about semantic commit messages [here](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716?permalink_comment_id=3867882).
```
[the remark this answers is not in the document]
Comment (rudrakshkarpe) [reply CONTRIBUTING.md:35]: Sure! Thanks for suggesting me

[the remark this answers is not in the document]
Comment (rudrakshkarpe) [reply CONTRIBUTING.md]: I've done all the suggested changes, please check if it's ready to get merged

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -26,3 +26,34 @@ information on using pull requests.
 
 This project follows [Google's Open Source Community
 Guidelines](https://opensource.google/conduct/).
+
+## Commit Message Conventions
+- Start with a short summary (50 characters or less) of the changes made.
+- Use the present tense and imperative mood.
+- Separate the summary from the body of the message with a blank line.
+- Use the body to explain what and why changes were made, as well as any necessary details.
+- Addionally, you can consider using [semantic commit messages](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716?permalink_comment_id=3867882) like "feat:","docs:",etc. which will provide additional context to the commit message.
```
Comment (ZetaTwo) [inline CONTRIBUTING.md:35]: space after comma.

**Review by ZetaTwo — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (mshudrak) [reply CONTRIBUTING.md:35]: @rudrakshkarpe this is not addressed yet.

Comment (rudrakshkarpe) [reply CONTRIBUTING.md]: > @rudrakshkarpe this is not addressed yet.

I apologize for the delay, I just fixed it. Please take a look at it.

**Comment by mshudrak** [comment]
LGTM

**Review by mshudrak — APPROVED** [review]

**Pull request merged**